eng/build.ps1

```diff
@@ -180,7 +180,7 @@ function Process-Arguments() {
   }
 
   if ($binaryLog -and ($binaryLogName -eq "")) {
-    $binaryLogName = "Build.binlog"
+    $script:binaryLogName = "Build.binlog"
```

Comment:
I'm scared.

Reply:
Well because you asked ... 

It's because when powershell reads a variable it will check function scope, then script scope. But if you write a variable it will write it in the function scope, even if one exists at script scope. So you need to be very explicit that you're writing a script scoped variable here.

Reply:
Should the `$binaryLogName` just above be `$script:binaryLogName` as well then?

Reply:
And the one on line 228?

Reply:
The uses on 182 are reads. Those will fall back to script scope. Line 228 is creating a new variable `$binaryLogPath` which is intentionally function level scope.

Why don't we always use `script:` when accessing script variables and instead rely on fallback rules? I don't have a good argument for that other than that's just kinda how it's written in a lot of cases. It's like asking why we require `this.` for extension methods but not for normal instance methods.